Tests/kaas/plugin/plugin_cluster_stacks.py

```diff
@@ -0,0 +1,366 @@
+
+            if result.returncode == 0:
+                pods_status = result.stdout.strip().splitlines()
+                for pod_status in pods_status:
+                    pod_info = pod_status.split()
+                    pod_name, phase, *readiness_states = pod_info
+
+                    # Check pod phase and all containers readiness
+                    if phase != "Running" or "false" in readiness_states:
+                        all_pods_ready = False
+                        logger.info(f"Pod {pod_name} in {cso_namespace} is not ready. Phase: {phase}, Ready: {readiness_states}")
+            else:
+                logger.error(f"Error fetching pods in {cso_namespace}: {result.stderr}")
+                all_pods_ready = False
+
+        except subprocess.CalledProcessError as error:
+            logger.error(f"Error checking pods in {cso_namespace}: {error}")
+            all_pods_ready = False
+
+        if all_pods_ready:
+            logger.info("All CSO pods in 'cso-system' namespace are ready.")
+            return True
+
+        logger.info("Waiting for CSO pods in 'cso-system' namespace to become ready...")
+        time.sleep(interval)
+
+    raise TimeoutError(f"Timed out after {timeout} seconds waiting for CSO pods in 'cso-system' namespace to become ready.")
+
+
+def wait_for_workload_pods_ready(namespace="kube-system", timeout=600, kubeconfig_path=None, max_retries=3, delay=30):
+    """
+    Waits for all pods in a specific namespace on a workload Kubernetes cluster to become ready.
+
+    :param namespace: The Kubernetes namespace where pods are located (default is "kube-system").
+    :param timeout: The timeout in seconds to wait for pods to become ready (default is 600).
+    :param kubeconfig_path: Path to the kubeconfig file for the target Kubernetes cluster.
+    :raises RuntimeError: If pods are not ready within the specified timeout.
+    """
+    for attempt in range(max_retries):
+        try:
+            kubeconfig_option = f"--kubeconfig {kubeconfig_path}" if kubeconfig_path else ""
+            wait_pods_command = (
+                f"kubectl wait -n {namespace} --for=condition=Ready --timeout={timeout}s pod --all {kubeconfig_option}"
+            )
+
+            # Run the command
+            subprocess.run(wait_pods_command, shell=True, check=True)
+            logger.info(f"All pods in namespace {namespace} in the workload Kubernetes cluster are ready.")
+            return True
+
+        except subprocess.CalledProcessError as error:
+            logger.warning(f"Attempt {attempt+1}/{max_retries} failed with error: {error}. Retrying in {delay} seconds...")
+            time.sleep(delay)
+
+    raise RuntimeError(f"Error waiting for pods in namespace '{namespace}' to become ready.")
+
+
+def load_config(config_path):
+    """
+    Loads the configuration from a YAML file.
+    """
+    if not os.path.exists(config_path):
+        raise FileNotFoundError(f"Configuration file {config_path} not found.")
```

Comment:
I'm pretty sure this is totally unnecessary, because the subsequent `open` will raise the same exception. This kind of checking also introduces the possibility of race conditions, because the `os.path.exists` can return `True`, but then the file can get deleted before the script arrives at `open`.

```suggestion
```